.github/workflows/main.yml

```diff
@@ -1,20 +1,84 @@
 name: CI
 
+# Events that trigger workflow
 on:
-  # Triggers the workflow on push or pull request events but only for the master branch
-  push:
-    branches: [ master ]
   pull_request:
-    branches: [ master ]
-  workflow_dispatch:
+  push:
+    branches:
+      - '**'
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-[a-z]+.[0-9]+'
 
+# Jobs can run sequentially or in parallel
 jobs:
-  build:
+  build-and-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.6', '3.7', '3.8']
+
+    steps:
+    - name: Checkout pull request HEAD commit instead of merge commit
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Check pull request commits for issue references
+      if: github.event_name == 'pull_request'
+      env:
+        GITHUB_PR_COMMITS_URL: ${{ github.event.pull_request.commits_url }}
+      run: |
+        sh .github/scripts/check_commit.sh
+
+    - name: Install source package and dependencies
+      run: |
+        python -m pip install -e .[dev] 
+
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```

Comment:
Why is `--max-line-length` set to 127?

Reply:
I pulled this from github's python workflow template... it is now fixed to say just `flake8 .` which uses our existing config file